code/modules/mob/living/bot/secbot.dm

```diff
@@ -510,6 +517,112 @@
 			secbot.next_destination = signal.data["next_patrol"]
 			secbot.closest_dist = dist
 
+/mob/living/bot/secbot/attack_hand(mob/living/carbon/human/M as mob)
+	..()
+
+	if(M.a_intent == I_HURT ) //assume he wants to hurt us.
+		idcheck = TRUE
+		target = M
+		mode = SECBOT_HUNT
+		var/mob/living/carbon/human/H = M
+		var/perpname = H.name
+		var/obj/item/weapon/card/id/id = H.GetIdCard()
+		if(id)
+			perpname = id.registered_name
+
+		var/datum/data/record/R = find_security_record("name", perpname)
+		if(R)
+			R.fields["criminal"] = "*Arrest*"
+		else
+			check_records = TRUE
+		broadcast_security_hud_message("[src] is under attack by <b>[target]</b>, [arrest_type ? "detaining" : "arresting"] a level [check_threat(target)] suspect in <b>[get_area(src)]</b>. Requesting backup", src)
+
+/mob/living/bot/secbot/attack_generic(var/mob/user, var/damage, var/attack_message)
+	..()
+
+	target = user
+	mode = SECBOT_HUNT
+	if(ishuman(user))
+		idcheck = TRUE
+		var/mob/living/carbon/human/H = user
+		var/perpname = H.name
+		var/obj/item/weapon/card/id/id = H.GetIdCard()
+		if(id)
+			perpname = id.registered_name
+
+		var/datum/data/record/R = find_security_record("name", perpname)
+		if(R)
+			R.fields["criminal"] = "*Arrest*"
+		else
+			check_records = TRUE
+		broadcast_security_hud_message("[src] is under attack by <b>[target]</b>, [arrest_type ? "detaining" : "arresting"] a level [check_threat(target)] suspect in <b>[get_area(src)]</b>. Requesting backup", src)
+
+/mob/living/bot/secbot/bullet_act(var/obj/item/projectile/P, var/def_zone)
+	..()
+
+	if (ismob(P.firer))
+		target = P.firer
+		mode = SECBOT_HUNT
+		if(ishuman(P.firer))
+			idcheck = TRUE
+			var/mob/living/carbon/human/H = P.firer
+			var/perpname = H.name
+			var/obj/item/weapon/card/id/id = H.GetIdCard()
+			if(id)
+				perpname = id.registered_name
+
+			var/datum/data/record/R = find_security_record("name", perpname)
+			if(R)
+				R.fields["criminal"] = "*Arrest*"
+			else
+				check_records = TRUE
+			broadcast_security_hud_message("[src] was shot with <b>[P]</b>, projectile came from <b>[target]</b>, [arrest_type ? "detaining" : "arresting"] a level [check_threat(target)] suspect in <b>[get_area(src)]</b>. Requesting backup", src)
```

Comment:
Another minor nitpick, there may be issues with super long range combat. For example, a sniper could hit the ed-209 unit and automatically identify someone despite being a long ways away.

Reply:
That is a good point

Reply:
Done, I made a check up there that checks if we can see mob that shot us.